crates/crypto/keystore/src/salsa.rs

```diff
@@ -0,0 +1,82 @@
+use ssz_types::{typenum::U16, FixedVector};
+
+fn rotate(value: u32, shift: u32) -> u32 {
+    (value << shift) | (value >> (32 - shift))
+}
+
+// Based on https://datatracker.ietf.org/doc/html/rfc7914#page-4
```

Comment:
```suggestion
/// Based on https://datatracker.ietf.org/doc/html/rfc7914#page-4
```

this should be a doc comment